src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -118,6 +118,18 @@ public Task WaitForResourceAsync(string resourceName, string? targetState = null
                                                      Justification = "targetState(s) parameters are mutually exclusive.")]
     public async Task<string> WaitForResourceAsync(string resourceName, IEnumerable<string> targetStates, CancellationToken cancellationToken = default)
     {
+        var appModel = _serviceProvider.GetService<DistributedApplicationModel>();
+
+        if (appModel is not null)
+        {
+            var resourceExist = appModel.Resources.Any(resource => resource.Name.Equals(resourceName, StringComparisons.ResourceName));
+
+            if (!resourceExist)
+            {
+                throw new InvalidOperationException($"Resource with name '{resourceName}' not found.");
```

Comment:
Should this only be done if `DefaultWaitBehaviour == WaitBehavior.StopOnResourceUnavailable`, as it is potentially possible that the resource could show up later.  (Perhaps if WaitBehavior is something else, it is worth logging something to highlight that the resoruce doesn't exist, even if it doesn't blow up completely)